cupy/cuda/memory.pyx

```diff
@@ -441,7 +442,38 @@ cdef class SingleDeviceMemoryPool:
             merged.next.prev = merged
         return merged
 
+    cpdef MemoryPointer _alloc(self, Py_ssize_t rounded_size):
+        hooks = get_memory_hooks()
+        if hooks:
+            hooks_values = hooks.values()
```

Comment:
I reverted not to use `six` for performance as function hook does. https://github.com/chainer/chainer/blob/beac2b52f5494ce16ae452f73033e2552840788e/chainer/function.py#L193